pygmt/clib/session.py

```diff
@@ -1530,8 +1527,13 @@ def virtualfile_from_data(  # noqa: PLR0912
         ...
         <vector memory>: N = 3 <7/9> <4/6> <1/3>
         """
-        kind = data_kind(
-            data, x, y, z, required_z=required_z, required_data=required_data
+        kind = data_kind(data, required=required_data)
+        validate_data_input(
+            data=data,
+            vectors=vectors,
+            names=names,
+            required_data=required_data,
+            kind=kind,
         )
```

Comment:
The validation checks have been moved from within `data_kind` to `virtualfile_from_data` here. But in `plot.py`, we actually use `data_kind` on its own here:

https://github.com/GenericMappingTools/pygmt/blob/3076ddcff5d21c6d5e06c9e588b5a785b6c3eec6/pygmt/src/plot.py#L217

Are we ok with raising GMTInvalidInput much later here in `virtualfile_from_data` (after all the keyword argument parsing), rather than early on in `data_kind`?